src/azure-cli/azure/cli/command_modules/servicefabric/custom.py

```diff
@@ -549,62 +552,68 @@ def remove_cluster_node(cmd, client, resource_group_name, cluster_name, node_typ
+    node_type_refs = [n for n in cluster.node_types if n.name.lower() == node_type.lower()]
+    if node_type_refs is None:
+        raise CLIError("Failed to find the node type in the cluster.")
+    node_type_ref = node_type_refs[0]
+    curr_node_type_durability = node_type_ref.durability_level
 
+    # get vmss extension durability
     compute_client = compute_client_factory(cli_ctx)
-    vmss = compute_client.virtual_machine_scale_sets.get(
-        resource_group_name, node_type)
-    fabric_exts = [ext for ext in vmss.virtual_machine_profile.extension_profile.extensions
-                   if ext.type.lower() == SERVICE_FABRIC_WINDOWS_NODE_EXT_NAME or ext.type.lower() == SERVICE_FABRIC_LINUX_NODE_EXT_NAME]
-    if fabric_exts is None:
-        raise CLIError("Failed to find service fabric extension")
-    fabric_ext = fabric_exts[0]
-    if fabric_ext.settings['durabilityLevel'] == durability_level:
-        return cluster
-    fabric_ext.settings['durabilityLevel'] = durability_level
-    fabric_ext.settings['enableParallelJobs'] = True
-    vmss_poll = compute_client.virtual_machine_scale_sets.create_or_update(
-        resource_group_name, vmss.name, vmss)
-
-    node_type = node_types[0]
-    node_type.durability_level = durability_level
-    patch_request = ClusterUpdateParameters(node_types=node_types)
-    update_cluster_poll = client.update(
-        resource_group_name, cluster_name, patch_request)
+    vmss = _get_cluster_vmss_by_node_type(compute_client, resource_group_name, cluster.cluster_id, node_type)
+    _get_sf_vm_extension(vmss)
+    fabric_ext_ref = _get_sf_vm_extension(vmss)
+    if fabric_ext_ref is None:
+        raise CLIError("Failed to find service fabric extension.")
+
+    curr_vmss_durability_level = fabric_ext_ref.settings['durabilityLevel']
+
+    # check upgrade
+    if curr_node_type_durability.lower() != curr_vmss_durability_level.lower():
+        logger.warning(
+            "The durability level is currently mismatched between the cluster ('%s') and the VM extension ('%s').",
+            curr_node_type_durability,
+            curr_vmss_durability_level)
+
+    # update cluster node type durability
+    if curr_node_type_durability.lower() != durability_level.lower():
+        node_type_ref.durability_level = durability_level
+        patch_request = ClusterUpdateParameters(node_types=cluster.node_types)
+        update_cluster_poll = client.update(resource_group_name, cluster_name, patch_request)
+        LongRunningOperation(cli_ctx)(update_cluster_poll)
+
+    # update vmss sf extension durability
+    if curr_vmss_durability_level.lower() != durability_level.lower():
+        fabric_ext_ref.settings['durabilityLevel'] = durability_level
+        fabric_ext_ref.settings['enableParallelJobs'] = True
+        vmss_poll = compute_client.virtual_machine_scale_sets.create_or_update(resource_group_name, vmss.name, vmss)
+        LongRunningOperation(cli_ctx)(vmss_poll)
```

Comment:
Change previous parallel works to series here ?

Reply:
yes, this was causing issues. The correct way to do it in sequence, first cluster upgrade and once it finishes do vmss update.